lib/adal.js

```diff
@@ -316,13 +320,43 @@ AuthenticationContext.prototype._renewIdToken = function (callback) {
     this.verbose('Navigate to:' + urlNavigate);
     this._saveItem(this.CONSTANTS.STORAGE.LOGIN_REQUEST, '');
     frameHandle.src = 'about:blank';
-    this._loadFrame(urlNavigate, 'adalIdTokenFrame');
+    this._loadFrameTimeout(urlNavigate, 'adalIdTokenFrame', this.config.clientId, this.CONSTANTS.LOADFRAME_TIMEOUT);
 };
 
 AuthenticationContext.prototype._urlContainsQueryStringParameter = function (name, url) {
     // regex to detect pattern of a ? or & followed by the name parameter and an equals character
     var regex = new RegExp("[\\?&]" + name + "=");
     return regex.test(url);
+};
+
+/* Calling _loadFrame but with a timeout to signal failure in loadframeStatus. Frames are left
+ * hanging when network errors occur.
+ * https://github.com/AzureAD/azure-activedirectory-library-for-js/issues/250.
+ * */
+AuthenticationContext.prototype._loadFrameTimeout = function (urlNavigation, frameName, resource, timeout) {
+    //set iframe session to pending
+    this.verbose('Set loading state to pending for: ' + resource);
+    window.loadframeStatus[resource] = this.CONSTANTS.LOADFRAME_PENDING;
+
+    this._loadFrame(urlNavigation, frameName);
+
+    var timeoutDuration = timeout/1000;
+    var resourceSaved = resource;
```

Comment:
Why do we need to save the resource locally? I see we access this value inside the timeout function, is the passed `resource` variable not available inside that method?